### PR TITLE
feat: added role assignments to groups

### PR DESCRIPTION
**Pull request by SanderBlom**
# Added role assignments to groups

## Description
Users should now be able to assign different RBAC roles to the group during creation. 

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Other

## Testing
See the testing [docs](../tests/readme.md) for more information about how to test your code 
- [x] Performed a local `terraform plan`, `apply`, and `destroy` to validate changes.
- [x] All tests are passing.
- [x] Added, updated, or removed tests when appropriate

## Checklist

Before submitting this PR, please ensure the following:

- [x] My commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) style guide
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings. I have ran `terraform validate` in the root module and all sub modules where I have made changes

## Additional Information

N.A

---

Thank you for contributing to Station!

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -175,6 +175,18 @@ variable "groups" {
       enabled = bool
       rule    = string
     }))
+    role_assignments = optional(map(object({
+      name                                   = optional(string)
+      scope                                  = optional(string) #Sets scope to current subscription if empty
+      role_definition_id                     = optional(string)
+      role_definition_name                   = optional(string) #Either this or the above must be provided
+      assign_to_workload_principal           = optional(bool)
```
Comment (kimfy) [inline variables.tf]: Is there a reason to include `assign_to_workload_principal` on a group role assignment?

File: variables.tf [hunk 2]
```diff
@@ -175,6 +175,18 @@ variable "groups" {
       enabled = bool
       rule    = string
     }))
+    role_assignments = optional(map(object({
+      name                                   = optional(string)
+      scope                                  = optional(string) #Sets scope to current subscription if empty
+      role_definition_id                     = optional(string)
+      role_definition_name                   = optional(string) #Either this or the above must be provided
```
Comment (kimfy) [inline variables.tf]: nitpick: comment is fluff

File: variables.tf [hunk 3]
```diff
@@ -175,6 +175,18 @@ variable "groups" {
       enabled = bool
       rule    = string
     }))
+    role_assignments = optional(map(object({
+      name                                   = optional(string)
+      scope                                  = optional(string) #Sets scope to current subscription if empty
```
Comment (kimfy) [inline variables.tf]: nitpick: comment is fluff

File: variables.tf [hunk 4]
```diff
@@ -175,6 +175,18 @@ variable "groups" {
...
+      role_definition_id                     = optional(string)
+      role_definition_name                   = optional(string) #Either this or the above must be provided
+      assign_to_workload_principal           = optional(bool)
+      condition                              = optional(string)
+      condition_version                      = optional(string)
+      delegated_managed_identity_resource_id = optional(string)
```
Comment (kimfy) [inline variables.tf]: `delegated_managed_identity_resource_id`: For a group, why is this necessary?

File: group/group.tf [hunk 5]
```diff
@@ -21,3 +21,19 @@ resource "azuread_group_member" "members" {
...
+  role_definition_name                   = each.value.role_definition_name
+  principal_id                           = azuread_group.group.object_id
+  condition                              = each.value.condition
+  condition_version                      = each.value.condition_version
+  delegated_managed_identity_resource_id = each.value.delegated_managed_identity_resource_id
+  description                            = each.value.description
+  skip_service_principal_aad_check       = each.value.skip_service_principal_aad_check == null ? false : each.value.skip_service_principal_aad_check
+}
+
+data "azurerm_client_config" "current" {}
```
Comment (kimfy) [inline group/group.tf]: Including `data "azurerm_client_config" "current" {}` in here is a bad idea and negates the previous PRs which aims to reduce the amount of data sources managed with Terraform Cloud.

I suggest this is resolved on caller level (`/groups.tf`)

File: group/group.tf [hunk 6]
```diff
@@ -21,3 +21,19 @@ resource "azuread_group_member" "members" {
   group_object_id  = azuread_group.group.object_id
   member_object_id = each.key
 }
+
+resource "azurerm_role_assignment" "roles" {
```
Comment (kimfy) [inline group/group.tf]: I think this resource should be its own file called `role_assignments.tf`.

File: group/outputs.tf [hunk 7]
```diff
@@ -6,6 +6,34 @@ output "display_name" {
   value = azuread_group.group.display_name
 }
 
+output "group_with_roles" {
```
Comment (kimfy) [inline group/outputs.tf]: I believe this adds a lot of overhead in terms of managing this output. I suggest just adding another output for role assignments:

```terraform
output "role_assignments" {
  value = azurerm_role_assignment.roles 
}
```

**Review by kimfy — CHANGES_REQUESTED** [review]

**Comment by SanderBlom** [comment]
@kimfy Can you verify the new changes?

**Comment by kimfy** [comment]
Resolves #106
Resolves #131

**Review by kimfy — APPROVED** [review]

**Pull request merged**